src/MCPClient/lib/archivematicaClient.py

```diff
@@ -21,173 +47,197 @@
+    supported_modules_config.read(file)
+    for client_script, client_script_path in supported_modules_config.items(
+            'supportedCommands'):
+        load_supported_modules_support(client_script, client_script_path)
     if django_settings.LOAD_SUPPORTED_COMMANDS_SPECIAL:
-        for key, value in supportedModulesConfig.items('supportedCommandsSpecial'):
-            loadSupportedModulesSupport(key, value)
+        for client_script, client_script_path in supported_modules_config.items(
+                'supportedCommandsSpecial'):
+            load_supported_modules_support(client_script, client_script_path)
+
+
+class ProcessGearmanJobError(Exception):
+    pass
+
+
+def _process_gearman_job(gearman_job, gearman_worker):
+    """Process a gearman job/task: return a 3-tuple consisting of a script
+    string (a command-line script with arguments), a task UUID string, and a
+    boolean indicating whether output streams should be captured. Raise a
+    custom exception if the client script is unregistered or if the task has
+    already been started.
+    """
+    # ``client_script`` is a string matching one of the keys (i.e., client
+    # scripts) in the global ``supported_modules`` dict.
+    client_script = gearman_job.task
+    task_uuid = str(gearman_job.unique)
+    logger.info('Executing %s (%s)', client_script, task_uuid)
+    data = cPickle.loads(gearman_job.data)
+    utc_date = getUTCDate()
+    arguments = data['arguments']
+    if isinstance(arguments, unicode):
+        arguments = arguments.encode('utf-8')
+    client_id = gearman_worker.worker_client_id
+    task = Task.objects.get(taskuuid=task_uuid)
+    if task.starttime is not None:
+        raise ProcessGearmanJobError({
+            'exitCode': -1,
+            'stdOut': '',
+            'stdError': 'Detected this task has already started!\n'
+                        'Unable to determine if it completed successfully.'})
+    task.client = client_id
+    task.starttime = utc_date
+    task.save()
+    client_script_full_path = supported_modules.get(client_script)
+    if not client_script_full_path:
+        raise ProcessGearmanJobError({
+            'exitCode': -1,
+            'stdOut': 'Error!',
+            'stdError': 'Error! - Tried to run an unsupported command.'})
+    replacement_dict['%date%'] = utc_date.isoformat()
+    replacement_dict['%jobCreatedDate%'] = data['createdDate']
+    # Replace replacement strings
+    for var, val in replacement_dict.items():
+        # TODO: this seems unneeded because the full path to the client
+        # script can never contain '%date%' or '%jobCreatedDate%' and the
+        # other possible vars have already been replaced.
+        client_script_full_path = client_script_full_path.replace(var, val)
+        arguments = arguments.replace(var, val)
+    arguments = arguments.replace('%taskUUID%', task_uuid)
+    script = client_script_full_path + ' ' + arguments
+    return script, task_uuid
```

Comment:
The docstring says that this function returns a 3-tuple of values, but here we are only returning the `script` and `task_uuid`. What happened to the capture output boolean, or was that aspirational? I think we should either update the docstring or add the missing boolean to the return value (and use it in `execute_command` in the call to `executeOrRun`), either way it needs changing.

Reply:
Good catch @lower29, I'll fix!

Reply:
> What happened to the capture output boolean, or was that aspirational?

It's something that Joel has been adding to the vanilla 1.8 WIP branch. I guess he forgot to delete it when he ported the refactor to 1.7.